base/util.jl

```diff
@@ -630,6 +633,7 @@ macro kwdef(expr)
     return quote
         $(esc(:($Base.@__doc__ $expr)))
         $kwdefs
+        $(esc(:($Base.kwdef_defaults)))(::Type{$(esc(S))}) = (;$(filter(p -> isexpr(p, :kw), parameters)...))
```

Comment:
Should this be `<: S`?

Reply:
Note that this changes what the macro returns:
```julia
julia> Base.@kwdef struct S  # without this PR, returns S
           a = 1
           b::Int = 2
           c
       end
S

julia> @kwdef struct S  # with this PR, returns kwdef_defaults
           a = 1
           b::Int = 2
           c
       end
kwdef_defaults (generic function with 3 methods)
```
I don't know whether we care, since without the macro a struct definition returns `nothing`
```julia
julia> struct S2  # without macro
           a
           b::Int
           c
       end

julia> ans === nothing
true

julia> struct S3  # without macto
           a
           b::Int
           c
           S3(a::Int,b::Int,c::Int) = new(a,b,c)
       end
```